internal/config/dockerlabels_test.go

```
@@ -0,0 +1,480 @@
+	allowList, found := refreshedAllowLists.FindByIP("172.20.0.2")
+	if !found {
+		t.Fatal("allowlist was not added for container IP")
+	}
+	if !matchAny(allowList.AllowedRequests[http.MethodGet], "/version") {
+		t.Fatal("refreshed allowlist does not contain the container label")
+	}
+}
+
+func TestRefreshDoesNotOverwriteNewerEventUpdate(t *testing.T) {
+	socketPath := filepath.Join(t.TempDir(), "docker.sock")
+	listener, err := net.Listen("unix", socketPath)
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)